script.js

```diff
@@ -1,40 +1,84 @@
-function colorClick(event) {
+function colorSelected({ target }) {
   const selectedColor = document.querySelector('.selected');
-  selectedColor.classList.remove('selected');
-  event.target.classList.add('selected');
-
-  /*
-    Não me recordava como adicionar uma classe e ao buscar no google encontrei:
-    https://pt.stackoverflow.com/questions/225809/como-adicionar-uma-classe-em-javascript-puro */
+  selectedColor.classList.toggle('selected');
+  target.classList.add('selected');
 }
 
-function pixelClick(event) {
-  const selectedColor = document.querySelector('.selected');
-  event.target.style.backgroundColor =
-    getComputedStyle(selectedColor).backgroundColor;
+function colorPixel(event) {
+  const selectedBackground = getComputedStyle(document.querySelector('.selected')).backgroundColor;
+  const { target } = event;
+  const currentPixelColor = getComputedStyle(target).backgroundColor;
 
-  /* 
-    https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle 
-  */
+  if (currentPixelColor === selectedBackground) {
+    target.style.backgroundColor = 'white';
+  } else {
+    target.style.backgroundColor = selectedBackground;
+  }
 }
 
-function clearColors(event) {
+function clearBoard() {
   const pixels = document.querySelectorAll('.pixel');
   for (let index = 0; index < pixels.length; index += 1) {
     pixels[index].style.backgroundColor = 'white';
   }
 }
 
-const colors = document.getElementsByClassName('color');
+function validateBoard(boardSize) {
+  if (boardSize < 5) return 5;
+  if (boardSize > 50) return 50;
+  if (Number.isNaN(boardSize)) return false;
+  return boardSize;
+}
 
+function resetPixelBoard(boardSize) {
+  const pixelBoard = document.getElementById('pixel-board');
+  pixelBoard.innerHTML = '';
+  pixelBoard.style.gridTemplateColumns = `repeat(${boardSize}, 40px)`;
+}
+
+function generateBoard() {
+  const boardSize = parseInt(document.getElementById('board-size').value, 10);
+
+  const validBoard = validateBoard(boardSize);
+  if (!validBoard) return alert('Board inválido!');
+  resetPixelBoard(validBoard);
+
+  for (let i = 0; i < validBoard; i += 1) {
+    for (let j = 0; j < validBoard; j += 1) {
+      const pixel = document.createElement('div');
+      pixel.classList.add('pixel');
+      pixel.style.backgroundColor = 'white';
+      pixel.addEventListener('click', colorPixel);
+      document.getElementById('pixel-board').appendChild(pixel);
+    }
+  }
```

Comment:
Por que dos `for` aninhados aqui? Pelo que entendi daria para fazer isso somente com 1 for ja que nao esta usando nenhuma das variaveis nas iteracoes.

Reply:
Se eu usar só um for, quando for alterado o tamanho do board ele gera com só uma linha. Pode-se dizer que o primeiro for gera as linhas e o segundo as colunas.

Reply:
Acho que entendi o que você quis dizer, fiz uns testes aqui e vou alterar.